### PR TITLE
behave consistently in [packagist] badges when called with no version arg

### LINKED ISSUE #2356: Invalid PHP version on Packagist
**Description:**

PHP version badge on Packagist returns `invalid` for some packages. It works normally for others.

**Working badges:**

PHP from Packagist - https://packagist.org/packages/filips123/monolog-phpmailer/

PHP from Packagist - https://packagist.org/packages/symfony/symfony/

**Invalid badges:**

PHP from Packagist - https://packagist.org/packages/filips123/config-writer

PHP from Packagist - https://packagist.org/packages/slim/slim/

### PR DESCRIPTION
**Pull request by chris48s**
Closes #2356 (although it does change behaviour by assuming we want a stable release by default instead of defaulting to an unstable branch alias).

Making this change doesn't do anything to address #2190, #1586 or #885 - all 3 badges still have the same behaviour in the case where a version like `1.0.0-ALPHA` or `1.0.0-rc` is used, but

`/packagist/l/namespace/package`
`/packagist/v/namespace/package`
`/packagist/php-v/namespace/package`

now all work out latest version in the same way instead of doing 3 different things.

### CODE REVIEW

File: services/packagist/packagist-helpers.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+'use strict'
+
+const {
+  compare: phpVersionCompare,
+  latest: phpLatestVersion,
+  isStable: phpStableVersion,
+} = require('../../lib/php-version')
+
+const getLatestVersion = function(versionsData, stable = true) {
```
Comment (paulmelnikow) [inline services/packagist/packagist-helpers.js]: Could you add some sazerac tests of this function? It's dense with logic so which makes it a little difficult to understand what its behavior is supposed to be.

File: services/packagist/packagist-helpers.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  // Map aliases (eg, dev-master).
+  const aliasesMap = {}
+  versions.forEach(version => {
+    const versionData = versionsData[version]
+    if (
+      versionData.extra &&
+      versionData.extra['branch-alias'] &&
+      versionData.extra['branch-alias'][version]
+    ) {
```
Comment (paulmelnikow) [inline services/packagist/packagist-helpers.js]: This is fine, though you could write the above even more functionally like this:

```js
const aliasesMap = {}
Object.entries(versionsData)
  .filter(
    (version, versionData) =>
      versionData.extra &&
      versionData.extra['branch-alias'] &&
      versionData.extra['branch-alias'][version]
  )
  .forEach((version, versionData) => {
    ...
```

**Comment by paulmelnikow** [comment]
@chris48s Would you be up for addressing these comments?

**Comment by chris48s** [comment]
I think I'm going to bail on this particular one :door: :running_man:

**Pull request closed without merging**